src/Foundation/Application.php

```diff
@@ -456,6 +540,24 @@ public function getCachedPackagesPath()
      */
     public function getCachedClassesPath()
     {
-        return $this->storagePath().'/framework/classes.php';
+        return $this->buildPath('/framework/classes.php', $this->storagePath());
+    }
+
+    /**
+     * Build path with portable directory seperator
+     *
+     * @param string $path
+     * @param string $prefix
+     * @return string
+     */
+    public function buildPath($path, $prefix = null)
+    {
+        if (!isset($prefix)) {
+            $prefix = $this->basePath;
+        }
+        $prefix = str_replace('/', DIRECTORY_SEPARATOR, $prefix);
+        $path = str_replace('/', DIRECTORY_SEPARATOR, $path);
+
+        return rtrim($prefix, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . ltrim($path, DIRECTORY_SEPARATOR);
```

Comment:
Why not use the new ::join() helper method here instead of duplicating the normalization work?